interpreter/cling/lib/Interpreter/MultiplexInterpreterCallbacks.h

```diff
@@ -139,6 +139,22 @@ namespace cling {
       for (auto&& cb : m_Callbacks)
         cb->ReturnedFromUserCode(StateInfo);
     }
+
+    void* LockCompilationDuringUserCodeExecution() override {
+      void* ret = nullptr;
+      for (auto&& cb : m_Callbacks) {
+        if (void* myret = cb->LockCompilationDuringUserCodeExecution()) {
+          assert(!ret && "Multiple state infos are not supported!");
+          ret = myret;
```

Comment:
odd pattern.  more than one callback can be executed but only one state is record(able).

Reply:
If you have multiple clients locking then you'll have a problem anyway. The `assert` is meant to signal that. I don't know of any other reasonable / cheap way of asserting that only one callback is dealing with locks, except for introducing a new synchronization callback type that doesn't offer a multiplexer. Maybe that's the way to go... in the future. I'll add a FIXME.